File: archetype-api/src/test/java/com/alo7/archetype/common/biz/mapper/BizAccountMapperTest.java
// Copyright 2019 Alo7 Inc. All rights reserved.

package com.alo7.archetype.common.biz.mapper;

import com.alo7.archetype.api.SpringBootArchetypeServer;
import com.alo7.archetype.api.config.DataSourceConfig;
import com.alo7.archetype.base.testing.BaseSpringTest;
import com.alo7.archetype.base.testing.database.MockDatabase;
import com.alo7.archetype.common.entity.biz.BizAccount;
import com.alo7.archetype.common.mapper.biz.BizAccountMapper;
import org.junit.Assert;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;

/**
 * @author Kelin Tan
 */
@SpringBootTest(webEnvironment = WebEnvironment.NONE, classes = SpringBootArchetypeServer.class)
@MockDatabase(name = DataSourceConfig.BIZ)
public class BizAccountMapperTest extends BaseSpringTest {
    @Autowired
    private BizAccountMapper bizAccountMapper;

    @Test
    public void testFindByAccount() {
        Assert.assertNull(bizAccountMapper.findByAccount("noAccount"));
        BizAccount account = bizAccountMapper.findByAccount("test1");

        Assert.assertNotNull(account);
        Assert.assertEquals("test1", account.getAccount());
        Assert.assertEquals("password1", account.getPassword());
        Assert.assertEquals("token1", account.getToken());
    }

    @Test
    public void testUpdateToken() {
        bizAccountMapper.updateToken(1L, "updateToken");

        Assert.assertEquals("updateToken", bizAccountMapper.findOne(1L).getToken());
    }
}

File: archetype-demo-database/src/test/java/com/kelin/archetype/database/BizAccountMapperTest.kt
// Copyright 2020 Kelin Inc. All rights reserved.

package com.kelin.archetype.database

import com.kelin.archetype.database.config.BizDatabase
import com.kelin.archetype.database.mapper.biz.BizAccountMapper
import com.kelin.archetype.test.KtBaseSpringTest
import com.kelin.archetype.test.database.MockDatabase
import org.junit.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import java.math.BigDecimal
import java.time.LocalDateTime

/**
 * @author Kelin Tan
 */
@MockDatabase(name = BizDatabase.NAME)
@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE)
class BizAccountMapperTest : KtBaseSpringTest() {
    @Autowired
    lateinit var bizAccountMapper: BizAccountMapper

    @Test
    fun `find by account`() {
        bizAccountMapper.findByAccount("test1") verify {
            account eq "test1"
            id eq 1
            password eq "password1"
            token eq "token1"
            age eq 10
            height eq 20
            money eq BigDecimal(100)
            birthDate eq LocalDateTime.of(2020, 1, 10, 15, 34, 27)
            verify eq true
        }
    }

    @Test
    fun `update token`() {
        bizAccountMapper.updateToken(1, "update")

        bizAccountMapper.findOne(1) verify {
            token eq "update"
        }
    }
}